src/types/xdg_activation_v1.zig

```diff
@@ -0,0 +1,51 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const XdgActivationV1 = extern struct {
+    // token timeout in milliseconds (0 to disable)
+    token_timeout_msec: u32,
+
+    tokens: wl.list.Head(XdgActivationTokenV1, "link"),
+
+    events: extern struct {
+        destroy: wl.Signal(*XdgActivationV1),
+        request_activate: wl.Signal(*XdgActivationV1.RequestActivate.event),
+    },
+
+    // Private state
+    global: *wl.Global,
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    pub const RequestActivate = extern struct {
```

Comment:
The pattern used elsewhere in zig-wlroots is to place this before the first field of `XdgActivationV1` and to do:
```zig
pub const event = struct {
    pub const RequestActivate = extern struct { ... };
};
```